common/endian.h

```diff
@@ -106,6 +106,20 @@
 			return result;
 		}
 	}
+
+// Test for GCC-compatible
+#elif GCC_ATLEAST(4, 8) || defined(__clang__)
```

Comment:
I wonder if this one shouldn't come first (before PSP one) because it's cleaner than writing a piece of assembly code.

Reply:
Forget about it, I just saw how it's done for 32-bits.

Reply:
It might still be worth assessing how useful the MIPS routines are with modern compilers - according to the [Compiler Explorer](https://godbolt.org/z/zsxdY9xaf), GCC 5.4 for MIPS at least already generates `wsbh`/`ror` instructions when using the GCC builtins, and for MIPS64r2 the assembly code looks like it'll be slower than the builtins because it doesn't use `dsbh`/`dshd` for `SWAP_BYTES_64`. I'm not sure about the special Allegrex instructions, though.

Reply:
For Allegrex, I will do some test but it should be OK according to the code:
https://github.com/pspdev/gcc/commit/1530c47ced6f7fc2a1479d74389b58ef0fb6fc36#diff-dad60b351908e704b73fd0b46c72674f8fc9fff6fea1900b902d8b04fae3fb1fR5871

Reply:
I can confirm that with latest PSP toolchain (GCC 11.2.0 allegrex), `__builtin_bswap32` generates `wsbw` and `__builtin_bswap16` generates `wsbh`.

Reply:
OK, I've added another commit that removes the MIPS versions.